compiler/src/iree/compiler/Codegen/Common/TransformDialectJitterPass.cpp

```diff
@@ -396,28 +111,15 @@ class TransformDialectJitterPass
   void runOnOperation() override {
     Operation *target = getOperation();
     LLVM_DEBUG(DBGS() << "input IR:\n" << *target);
-
-    // TODO: connect the config above to the reduction below.
-    // This connects to Codegen/LLVMGPU/KernelConfig.cpp
-    //   LogicalResult initGPULaunchConfig(ModuleOp moduleOp)
-    //     LogicalResult setWarpReductionConfig(func::FuncOp, LinalgOp)
-    MLIRContext *ctx = target->getContext();
-    Location loc = target->getLoc();
-
-    OpBuilder b(ctx);
-    auto topLevelTransformModule = b.create<ModuleOp>(loc);
-    Region &topLevelTransformRegion = topLevelTransformModule.getBodyRegion();
-    b.setInsertionPointToStart(&topLevelTransformRegion.front());
-    b.create<::transform_ext::CanonicalizedSequenceOp>(
-        target->getLoc(), transform::FailurePropagationMode::Suppress,
-        [](OpBuilder &b, Location loc, Value variantH) {
-          ImplicitLocOpBuilder ib(loc, b);
-          buildReductionCudaStrategy(ib, variantH);
-          b.create<transform::YieldOp>(loc);
+    Region *topLevelTransformRegion = nullptr;
+    target->walk(
+        [&](::transform_ext::CanonicalizedSequenceOp transformTopLevelOp) {
```

Comment:
A more robust way to do this: find the op with `PossibleTopLevelTransformOpTrait` that has no operands.

Reply:
ah thanks, I was looking for the right config. I'll clean this up